node-src/git/getChangedFilesWithReplacement.ts

```diff
@@ -39,20 +39,41 @@ export async function getChangedFilesWithReplacement(
       `Got error fetching commit for #${build.number}(${build.commit}): ${err.message}`
     );
 
-    if (/(bad object|uncommitted changes)/.test(err.message)) {
-      const replacementBuild = await findAncestorBuildWithCommit(ctx, build.number);
+    if (!/(bad object|uncommitted changes)/.test(err.message)) {
+      throw err;
+    }
+
+    // Try to find a replacement build by checking multiple ancestor builds
+    let skip = 0;
+    const page = 10;
+    const limit = 80;
 
-      if (replacementBuild) {
+    while (skip < limit) {
+      const replacementBuild = await findAncestorBuildWithCommit(ctx, build.number, {
+        page,
+        limit: Math.min(page, limit - skip),
+      });
+
+      if (!replacementBuild) {
+        break;
+      }
+
+      try {
         ctx.log.debug(
           `Found replacement build for #${build.number}(${build.commit}): #${replacementBuild.number}(${replacementBuild.commit})`
         );
         const changedFiles = (await getChangedFiles(replacementBuild.commit)) || [];
```

Comment:
So if I understand correctly, this is the function that times out. Why would it time out? Would each ancestor build would have the same issue?